qCC/ccGraphicalTransformationTool.cpp

```diff
@@ -861,6 +913,49 @@ void ccGraphicalTransformationTool::glRotate(const ccGLMatrixd& rotMat)
 	updateAllGLTransformations();
 }
 
+void ccGraphicalTransformationTool::incrementalTransform() {
+	
+	//We check the sender to handle forward or backward transformation
+	QObject* senderButton = sender();
+	const bool isForwardTransform = senderButton == incrementalForwardButton;
+	
+	// The rotation part:
+	// No incremental rotation if rotComboBoxItems == XYZ or rotComboBoxItems == NONE because it has no sense
+	// it could be handled in the subsequent switch cases but it is somehiow more explicit to check
+	// if the SpinBox is enabled
+	if(incrementalRotSpin->isEnabled()) {
+		const double alpha = isForwardTransform ? CCCoreLib::DegreesToRadians(incrementalRotSpin->value()) : -CCCoreLib::DegreesToRadians(incrementalRotSpin->value());
+		if (m_advMode && m_advRotateRef != nullptr)
+		{
+			m_rotation = m_rotation * arbitraryVectorRotation(alpha, m_advRotationAxis);
+		} else {
+			ccGLMatrixd rotMat;
+			// compute rotation matrix from spinbox
+			switch ((rotComboBoxItems)rotComboBox->itemData(rotComboBox->currentIndex()).toInt())
+			{
+			case rotComboBoxItems::X:
+				rotMat.initFromParameters(alpha, CCVector3f(1.f,0.f,0.f), CCVector3f(0.f,0.f,0.f));
```

Comment:
since rotMat is a ccGLMatrixd, I would expect you to init it with CCVector3d vectors?

(same for Y and Z)

Reply:
Yep, good catch! it's an artifact of a previous attempt to integrate this feature.